### PR TITLE
feat: Setup NX monorepo

### PR DESCRIPTION
**Pull request by PoProstuMieciek**
Apps:
- api (Nest.js)
- client + client-e2e (React, Typescript)

Libs:
- common (Typescript)

### CODE REVIEW

**Review by KonTra7cf — APPROVED** [review]

**Review by Mariusz-web — APPROVED** [review]

**Comment by Veryyapeee** [comment]
What is client-e2e for? I bet we won't have e2e tests on frontend, no time fot that, that workapce is kinda useless

Readme is missing, it's auto generated, we need some more information, for example which is which in each workspace, commands etc.

Also get rid of jest, as before, we won't have any tests xD (just in the root, in for example client workspace would be cleaned by the frontend team)

Besides that, looks cool to me, kinda huge boilerplate created by that NX, not gonna lie, but I think that after some cleanup it would look cool

**Comment by Veryyapeee** [comment]
Also we have some useless packages, like for example rxjs which we wouldn't be using,

Also linting scripts are missing

I created separate branch (i wanted to play with nx, never used that before :P ) on which I cleanup the project from tests and useless libraries, you can check it out and use that

https://github.com/coders-camp-2021-best-team/FitaTAM/tree/issue-3-feat/app-setup-monorepo-cleanup-tests
`issue-3-feat/app-setup-monorepo-cleanup-tests`

File: package.json [hunk 1]
```diff
@@ -18,44 +19,32 @@
         "react-router-dom": "6.3.0",
         "reflect-metadata": "^0.1.13",
         "regenerator-runtime": "0.13.7",
-        "rxjs": "^7.0.0",
```
Comment (PoProstuMieciek) [inline package.json]: bruh

Comment (PoProstuMieciek) [reply package.json]: do you know how much time i wasted figuring out that someone removed that package xDDD [look at this issue](https://github.com/nestjs/nest/issues/7468#issuecomment-876174870)

**Pull request merged**